src/importer/storage.ts

```diff
@@ -0,0 +1,96 @@
+import { ImporterState } from './types';
+import { SheetDefinition } from '../sheet/types';
+
+const DB_NAME = 'HelloCSV';
+const DB_VERSION = 1;
+const STORE_NAME = 'state';
+
+export async function getIndexedDBState(
+  sheetDefinitions: SheetDefinition[],
+  customKey?: string | null
+): Promise<ImporterState | null> {
+  return new Promise((resolve, reject) => {
+    const key = stateKey(sheetDefinitions, customKey);
+    const request = indexedDB.open(DB_NAME, DB_VERSION);
+
+    request.onerror = () => reject(request.error);
+    request.onsuccess = () => {
+      const db = request.result;
+      const transaction = db.transaction(STORE_NAME, 'readonly');
+      const store = transaction.objectStore(STORE_NAME);
+      const getRequest = store.get(key);
+
+      getRequest.onerror = () => resolve(null);
+      getRequest.onsuccess = () => {
+        try {
+          const result = getRequest.result;
+          result.sheetDefinitions = sheetDefinitions;
+          resolve(result);
+        } catch (error) {
+          resolve(null);
+        }
+      };
+    };
+
+    request.onupgradeneeded = (event) => {
+      const db = (event.target as IDBOpenDBRequest).result;
+      const oldVersion = event.oldVersion;
+
+      // Clear data if upgrading from an older version
+      if (oldVersion < DB_VERSION) {
```

Comment:
Is this if needed?

Reply:
just in case we upgrade the DB for any reason 🤷🏼

Reply:
Yes, but what would be the case that this method is called and `oldVersion >= DB_VERSION`? Is it possible?

Reply:
hmm maybe you're right

I'll remove this and only add something else if we do upgrade our database

Reply:
Just for clarity - I think we can remove the if - but leave it's body - just call it anytime

Reply:
I think you removed too much 😓 - we still want to recreate the DB on each version upgrade IMO

Reply:
shouldn't we add this functionality when we do need to upgrade? Right now it will always be on version 1 unless a new PR upgrades the DB version

Reply:
I think we could add it already. 
1. Don't you need to have it for the initial version when DB doesn't exist yet on the user end?
2. I think the implementation would always be the same  - bumping version means that there is some breaking change in the state (I don't think we will bump version for minor state changes - we could write in in `deployment.mdx` file - to make sure DB_VERSION is bumped in case of breaking changes in ImporterState) and hence the db should be dropped - we don't want to risk having incompatible types in there